hikari/impl/special_endpoints.py

```diff
@@ -1638,9 +1639,11 @@ class SelectMenuBuilder(special_endpoints.SelectMenuBuilder[_ContainerProtoT]):
     """Builder class for select menus."""
```

Comment:
Did you consider having type specialised builders (and keeping select menu builder as the base class) in the same style as button instead to avoid the need for those untested raise cases in the new set and add methods